library/network/src/modules/SuSEFirewall.rb

```diff
@@ -33,7 +33,18 @@
 require "yast"
 
 module Yast
-  class SuSEFirewallClass < Module
+  # Factory for construction of appropriate firewall object based on
+  # desired backend.
+  class Firewall < Module
```

Comment:
Classes that are in the `modules` directory and inherit from `Yast::Module` are expected to be named `FooClass`, and have a single instance named `Foo`. This one breaks the rule.
On the other hand, I see that you somehow need to keep the old API and dispatch between the 2 backends.